### PR TITLE
Add some comments and types

### PR DESCRIPTION
**Pull request by tonycaputome**
- Rename model folder to types.
- Add some comment.
- Change tsconfig allows TS look for a index file in a folder.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,11 +1,20 @@
-import { arraify, isPromise } from "./lib/utils";
-import { CacheItem } from "./models/CacheItem";
-
-
-
+import { arraify, isPromise } from "./lib";
+import { CacheItem, TKey, TMapCache, TStrings } from "./types";
+
```
Comment (Cadienvan) [inline src/index.ts:3]: I completely agree with you, still I don't have the knowledge to provide a working example. Please, remove the comment or change it as a "@todo"

File: src/index.ts [hunk 2]
```diff
@@ -14,33 +23,31 @@ export class KeyValueCache {
...
 
-  #reconstructMapKey(keys: string) {
+  #reconstructMapKey(keys: string): TStrings {
     return keys.split(this.KEY_SEPARATOR);
   }
 
   exec(
-    fn: Function,
-    key: string | Array<string>,
+    fn: () => Promise<any>,
```
Comment (Cadienvan) [inline src/index.ts]: It doesn't necessarily have to be a promise, you can pass a long running sync function. Please, revert to a sync function or bring both sync and async to the table.

File: src/index.ts [hunk 3]
```diff
@@ -14,33 +23,31 @@ export class KeyValueCache {
...
+    key: TKey,
     threshold = 1,
-    dependencyKeys: string | Array<string> = [],
+    dependencyKeys: TKey = [],
     ttl = this.DEFAULT_TTL
-  ): unknown {
+  ): Pick<CacheItem, "value"> | Promise<any> {
```
Comment (Cadienvan) [inline src/index.ts]: Thank you, that was exactly the static typing I was looking for!

File: src/index.ts [hunk 4]
```diff
@@ -50,33 +57,38 @@ export class KeyValueCache {
...
       currentInvalidations: 0,
-      ttl
+      ttl,
     });
   }
 
-  get(key: string | Array<string>) {
-    const _keys = arraify(key)
-    const cacheDataItem = this.#appCache.get(this.#getMapKey(_keys));
+  isCached = (key: TKey) => {
```
Comment (Cadienvan) [inline src/index.ts]: Not sure about this method, prepending with an "is" is supposedly expected to return a boolean, but this is not. Still, not gonna argue about that, just wanted some discussion.

File: src/index.ts [hunk 5]
```diff
@@ -138,16 +148,17 @@ export class KeyValueCache {
     return count;
   }
 
-  invalidateByKeys(keys: Array<string | RegExp>): number {
-    return keys.reduce((acc, key) => acc + this.invalidateByKey(key), 0);
+  invalidateByKeys(keys: Array<string | RegExp>) {
```
Comment (Cadienvan) [inline src/index.ts:149]: A counter could be useful for debugging purposes, who knows what the use case could be. Please, revert back to returning the sum.

**Review by Cadienvan — CHANGES_REQUESTED** [review]
Thank you, I just put a couple things to consider / revert, but really a lot of thanks for the invaluable help!

File: src/index.ts (hunk 4 above)
Comment (tonycaputome) [reply src/index.ts]: You are right, usually prepending "is" means we are waiting a return value as boolean. I just created this method due the replicated code. Just that
```ts
const _keys = arraify(key);
return this.#appCache.get(this.#getMapKey(_keys));
```

File: src/index.ts (hunk 5 above)
Comment (tonycaputome) [reply src/index.ts:149]: No problem.

**Comment by Cadienvan** [comment]
Thanks a lot!

**Pull request merged**